smithy-codegen-core/src/test/java/software/amazon/smithy/codegen/core/ArtifactMetadataTest.java

```diff
@@ -0,0 +1,42 @@
+package software.amazon.smithy.codegen.core;
+
+import org.junit.jupiter.api.Test;
+import software.amazon.smithy.model.node.ObjectNode;
+
+class ArtifactMetadataTest {
+
+    @Test
+    void toNode() {
+        ArtifactMetadata am = new ArtifactMetadata();
+        am.setId("a");
+        am.setVersion("b");
+        am.setType("c");
+        am.setTimestamp("d");
+
+        ObjectNode node = am.toNode();
+
+        assert node.getStringMember(ArtifactMetadata.ID_TEXT).get().getValue().equals("a");
```

Comment:
Our tests use the `hamcrest` or `junit` assertions and matchers. You can find examples [here](https://github.com/awslabs/smithy/blob/42faf2f01322ad4adb1cae915b4daec08e0b19b5/smithy-model/src/test/java/software/amazon/smithy/model/node/NodeTest.java#L159), [here](https://github.com/awslabs/smithy/blob/bbb166dabafd2da7bfe18c1fbd1e1713ec96a2f9/smithy-utils/src/test/java/software/amazon/smithy/utils/SimpleParserTest.java#L17), and [here](https://github.com/awslabs/smithy/blob/42faf2f01322ad4adb1cae915b4daec08e0b19b5/smithy-model/src/test/java/software/amazon/smithy/model/node/NodeTest.java#L144).